### PR TITLE
Cluster conditions evaluate against  operator instead of release version

### PR DESCRIPTION
**Pull request by sslavic**
Towards fixing giantswarm/giantswarm#7465

This is cluster-operator part of the fix. For aws-operator part of the fix see https://github.com/giantswarm/aws-operator/pull/2064

### CODE REVIEW

File: service/controller/clusterapi/v22/resource/clusterstatus/resource.go [hunk 1]
```diff
@@ -16,13 +16,17 @@ type Config struct {
...
+	Provider string
 }
 
 type Resource struct {
 	accessor  Accessor
 	cmaClient clientset.Interface
 	g8sClient versioned.Interface
 	logger    micrologger.Logger
+
+	provider string
```
Comment (tomahawk28) [inline service/controller/clusterapi/v22/resource/clusterstatus/resource.go]: Can it be enum type? so don't have it as a string?

Comment (sslavic) [reply service/controller/clusterapi/v22/resource/clusterstatus/resource.go]: It's consistent with other resources. If agreed we can refactor all in separate PR.

Comment (tomahawk28) [reply service/controller/clusterapi/v22/resource/clusterstatus/resource.go]: I think it's better to have it in other PRs.

Comment (rossf7) [reply service/controller/clusterapi/v22/resource/clusterstatus/resource.go]: We don't typically have enums for these flags and it is set via a configmap which is templated by draughtsman.

https://github.com/giantswarm/cluster-operator/blob/14bd41618c35eb3fe24b34e269a8d167b752bc61/helm/cluster-operator-chart/templates/configmap.yaml#L42

What is the benefit of using an enum here?

Comment (tomahawk28) [reply service/controller/clusterapi/v22/resource/clusterstatus/resource.go]: if it's coming by configmap resource, I don't think it's really necessary to have it as enum.

File: service/controller/clusterapi/v22/resource/clusterstatus/create.go [hunk 2]
```diff
@@ -116,12 +116,14 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 	return nil
 }
 
-func (r *Resource) computeClusterConditions(ctx context.Context, cluster cmav1alpha1.Cluster, clusterStatus g8sv1alpha.CommonClusterStatus, nodes []corev1.Node, machineDeployments []cmav1alpha1.MachineDeployment) g8sv1alpha.CommonClusterStatus {
+func (r *Resource) computeClusterConditions(ctx context.Context, cc *controllercontext.Context, cluster cmav1alpha1.Cluster, clusterStatus g8sv1alpha.CommonClusterStatus, nodes []corev1.Node, machineDeployments []cmav1alpha1.MachineDeployment) g8sv1alpha.CommonClusterStatus {
```
Comment (kopiczko) [inline service/controller/clusterapi/v22/resource/clusterstatus/create.go]: You should be able to extract controllercontext from context easily. I wouldn't pass it as argument.

Comment (sslavic) [reply service/controller/clusterapi/v22/resource/clusterstatus/create.go]: extracting controllercontext (again) can throw error, then have to change signature of the function to throw error, and handle it from call site - seemed like lots of noise for no obvious value, but will trust your guidance

Comment (kopiczko) [reply service/controller/clusterapi/v22/resource/clusterstatus/create.go]: I see. I didn't realize it doesn't return error 👍

**Comment by sslavic** [comment]
Tested on gauss

**Comment by sslavic** [comment]
> Change is fine. Is there any need to check if existing `seal` clusters need some manual patches before applying this?

@puja108 @rossf7 correct me if wrong, I thought communication was that there would be no upgrade support for these pre-release node pool clusters.

**Comment by rossf7** [comment]
> correct me if wrong, I thought communication was that there would be no upgrade support for these pre-release node pool clusters.

@sslavic Yes my understanding was we deployed an initial release to `seal` for testing. But there would be a 2nd release that would be upgradable. 

But maybe I'm wrong or that has changed. @puja108 Maybe you have more info?

**Comment by sslavic** [comment]
@J-K-C @puja108 what's your take, do we have to support upgrading of early access node pool clusters or not?

**Comment by sslavic** [comment]
@pipo02mix suggests we're all clear, no upgrade support needed https://gigantic.slack.com/archives/C63SJF71D/p1574208651004300?thread_ts=1574071139.003600&cid=C63SJF71D

**Pull request merged**